src/Simplify/AstHelpers.elm

```diff
@@ -861,11 +861,12 @@ If desired, call in combination with `qualify`
 -}
 qualifiedToString : ( ModuleName, String ) -> String
 qualifiedToString ( moduleName, name ) =
-    if List.isEmpty moduleName then
-        name
+    case moduleName of
+        [] ->
+            name
 
-    else
-        moduleNameToString moduleName ++ "." ++ name
+        moduleNamePart0 :: moduleNamePart1Up ->
+            moduleNameToString (moduleNamePart0 :: moduleNamePart1Up) ++ "." ++ name
```

Comment:
Not fond of changes like these where we deconstruct lists to reconstruct them right after, mostly for performance reasons :thinking:

Reply:
Ok, used the variable we cased on instead of reconstructing. Add a comment if you find more cases like that.

(All these again to me seem optimizable by some tool. Kinda sad that we have to make these micro-optimizations here.)

Reply:
(not going to disagree with you here :/ )